lib/commands/stack/ssh.js

```diff
@@ -109,11 +108,28 @@ function connect( data ) {
 		}
 
 		const message = reason ? `${ reason }. Disconnected.` : 'Disconnected.';
-		process.stderr.write( message );
+		if ( ! porcelain ) {
+			process.stdout.write( message );
+		}
 		session.close();
 		process.exit();
 	} );
+
+	let firstOutput = true;
 	session.on( 'output', ( data ) => {
+
+		// If this is the first output chunk and we're in porcelain mode, strip the initial 2 lines
+		// this contain somethign like:
+		//     exec sudo sandbox-exec 'clear; $command'
+		//     sh-4.2$ exec sudo sandbox-exec 'clear; $command'
+		if ( porcelain && firstOutput ) {
+			data = data.split( '\n' ).slice(2).join('\n').trim();
+		}
```

Comment:
Something you can do here is use the `clear` itself; it outputs the ANSI control characters for clearing, so you can split on that. Even cleaner, change the backend command to output just the one code with something like

```
exec sudo sandbox-exec "printf '\033[2J'; $command"
```

And then here, `data.split( '033[2J', 2 )[1]`